src/seat/mod.rs

```diff
@@ -102,6 +105,38 @@ impl SeatState {
         self.get_pointer_with_data(qh, seat, PointerData::default())
     }
 
+    /// Creates a pointer from a seat with the provided theme.
+    ///
+    /// ## Errors
+    ///
+    /// This will return [`SeatError::UnsupportedCapability`] if the seat does not support a pointer.
+    pub fn get_pointer_with_theme<D>(
```

Comment:
Seems like we also need a `get_pointer_with_theme_and_data`?

These `with_xyz` functions don't scale very well...